---
title: "OXR Targeted Sample Transcripts"
output: pdf_document
date: "2023-10-13"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

This code is meant to take the transcripts that were caught from the metatranscript data, look at the Differential expression analysis, include the counts and the TPM, along with the KO information and annotations and make a single table.

```{r}
## OXR-9
###-------- load packages you need --------###
library(DESeq2)
# library(DEVis)
library(dplyr)

###-------- load data process data --------###
dta <- read.csv("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-9/OXR-9_expt.ct.txt", sep = "\t",
                header=TRUE, row.names = "gene_id")

# remove rows with all zeros
dta <- dta[rowSums(dta[])>0,]
sort(colSums(dta))
sum(colSums(dta))

###-------- define data structure --------###
samples <- data.frame(row.names=c("C1B11F1","C1B11F2", "C1B11F3", # This looks like an outlier (9k, 3k, v 300)
                                  "C1B12F1","C1B12F2","C1B12F3","C2B1F1","C2B1F2",
                                  "C2B1F3"), 
                      condition=as.factor(c(rep("C1B11",3),
                                            # rep("C1B12",3),rep("C2B1",3))))
                                            rep("Out",6))))
### Get all samples in the same order
all(rownames(samples) %in% colnames(dta))
all(rownames(samples) == colnames(dta))
dta <- dta[, rownames(samples)]
all(rownames(samples) == colnames(dta))
### Make data integers
dta2 <- as.data.frame(lapply(dta, as.integer))
rownames(dta2) <- rownames(dta)
dta2 <- dta2[rowSums(dta2) > 0,] # Remove more rows with 0's
colSums(dta)
###-------- create deseq object and process --------###
dta2["fakey",] <- rep(1, ncol(dta2))

cds <- DESeqDataSetFromMatrix(countData = dta2, colData=samples, design=~condition)
cds1 <- DESeq(cds)

ref <- results(cds1)
ref[order(ref$padj),]

c1b11vOut <- results(cds1, contrast=c("condition","C1B11","Out"))
data <- as.data.frame(c1b11vOut[order(c1b11vOut$padj),])
# View(as.data.frame(c1b11vOut[order(c1b11vOut$padj),]))

data <- subset(data, padj < 0.1)

output.file <- file("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-9/diff_expr_analysis/OXR-9_DESeq_padj.txt", "wb")
write.table(data, file = output.file, row.names = T, col.names = T)
close(output.file)

ids <- row.names(data) %>% as.factor()
ids <- as.factor(ids)

output.file <- file("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-9/diff_expr_analysis/OXR-9_padj_trans.txt", "wb")
write.table(ids, file = output.file, row.names = F, col.names = F)
close(output.file)

# sed -ie 's/"//g' OXR-9_padj_trans.txt
# copy paste from the padj_trans file to new file in Notepad++ to avoid windows v linux issues.
# grep -f file.txt out.emapper.decorated.gff > OXR-9_diff_expr.txt

#####
# Add TPM data to the "data" data frame
tpm <- read.csv("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-9/OXR-9_TPM.txt", sep = "\t",
                header=TRUE, row.names = "gene_id")
tpm2 <- tpm[,c("C1B11F1","C1B11F2", "C1B11F3")] 
tpm3 <- tpm2[row.names(tpm2) %in% row.names(data),]
names(tpm3) <- c("TPM-C1B11F1","TPM-C1B11F2","TPM-C1B11F3")

# Add the count Data 
dta2 <- dta[,c("C1B11F1","C1B11F2", "C1B11F3")] 
dta3 <- dta2[row.names(dta2) %in% row.names(data),]
names(dta3) <- c("COUNT-C1B11F1","COUNT-C1B11F2","COUNT-C1B11F3")

new <- cbind(data[sort(row.names(data)),],tpm3[sort(row.names(tpm3)),],dta3[sort(row.names(dta3)),])

# Can't do this until the "OXR-X_padj_trans.txt" is run through the bash script "DESeq_diff_expr_from_eggNOG.sh" which collects the differentially expressed genes with the eggNOG annotations

codex <- read.csv("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-9/diff_expr_analysis/OXR-9_eggNOG_codex.txt", sep = " ", header = F, row.names = 1)

new2 <- cbind(data[sort(row.names(data)),],tpm3[sort(row.names(tpm3)),],dta3[sort(row.names(dta3)),],codex[sort(row.names(codex)),])

colnames(new2)[13] <- "eggNOG_alt_name"

new2$eggNOG_name <- row.names(new2)

new2 <- new2[,c("eggNOG_name","eggNOG_alt_name","baseMean","log2FoldChange","lfcSE","stat","pvalue","padj","TPM-C1B11F1","TPM-C1B11F2","TPM-C1B11F3","COUNT-C1B11F1","COUNT-C1B11F2","COUNT-C1B11F3")]

# Load the eggNOG annotation file that was parsed from the "DESeq_diff_expr_from_eggNOG.sh" script
anno <- read.csv("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-9/diff_expr_analysis/OXR-9_diff_expr_annotations.txt", sep = "\t", header = T)

all <- left_join(new2, anno, by = join_by(eggNOG_alt_name == X.query))

output.file <- file("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-9/diff_expr_analysis/OXR-9_annos_w-ALL.txt", "wb")
write.table(all, file = output.file, row.names = F, col.names = T, sep = "\t")
close(output.file)
```


```{r}
## OXR-11
###-------- load packages you need --------###
library(DESeq2)
# library(DEVis)
library(dplyr)

###-------- load data process data --------###
dta <- read.csv("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-11/OXR-11_expt.ct.txt", sep = "\t",
                header=TRUE, row.names = "gene_id")

# remove rows with all zeros
dta <- dta[rowSums(dta[])>0,]
sort(colSums(dta))
sum(colSums(dta))

###-------- define data structure --------###
samples <- data.frame(row.names=c("C1B3F1","C1B3F3", "C1B3F2", 
                                  "C1B12F1","C1B12F2","C1B12F3",
                                  "C2B1F1","C2B1F2","C2B1F3"), 
                      condition=as.factor(c(rep("C1B3",3),
                                            # rep("C1B12",3),rep("C2B1",3))))
                                            rep("Out",6))))
### Get all samples in the same order
all(rownames(samples) %in% colnames(dta))
all(rownames(samples) == colnames(dta))
dta <- dta[, rownames(samples)]
all(rownames(samples) == colnames(dta))
colSums(dta)
### Make data integers
dta2 <- as.data.frame(lapply(dta, as.integer))
rownames(dta2) <- rownames(dta)
dta2 <- dta2[rowSums(dta2) > 0,] # Remove more rows with 0's
###-------- create deseq object and process --------###
dta2["fakey",] <- rep(1, ncol(dta2))

cds <- DESeqDataSetFromMatrix(countData = dta2, colData=samples, design=~condition)
cds1 <- DESeq(cds)

ref <- results(cds1)
ref[order(ref$padj),]

c1b3vOut <- results(cds1, contrast=c("condition","C1B3","Out"))
data <- as.data.frame(c1b3vOut[order(c1b3vOut$padj),])
# View(as.data.frame(c1b3vOut[order(c1b3vOut$padj),]))

data <- subset(data, padj <= 0.1)

output.file <- file("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-11/diff_expr_analysis/OXR-11_DESeq_padj.txt", "wb")
write.table(data, file = output.file, row.names = T, col.names = T)
close(output.file)


ids <- row.names(data) %>% as.factor()
ids <- as.factor(ids)

output.file <- file("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-11/diff_expr_analysis/OXR-11_padj_trans.txt", "wb")
write.table(ids, file = output.file, row.names = F, col.names = F)
close(output.file)

#####
# Add TPM data to the "data" data frame
tpm <- read.csv("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-11/OXR-11_TPM.txt", sep = "\t",
                header=TRUE, row.names = "gene_id")
tpm2 <- tpm[,c("C1B3F1", "C1B3F2","C1B3F3")] 
tpm3 <- tpm2[row.names(tpm2) %in% row.names(data),]
names(tpm3) <- c("TPM-C1B3F1","TPM-C1B3F2","TPM-C1B3F3")

# Add the count Data 
dta2 <- dta[,c("C1B3F1","C1B3F2", "C1B3F3")] 
dta3 <- dta2[row.names(dta2) %in% row.names(data),]
names(dta3) <- c("COUNT-C1B3F1","COUNT-C1B3F2","COUNT-C1B3F3")

new <- cbind(data[sort(row.names(data)),],tpm3[sort(row.names(tpm3)),],dta3[sort(row.names(dta3)),])

# Can't do this until the "OXR-X_padj_trans.txt" is run through the bash script "DESeq_diff_expr_from_eggNOG.sh" which collects the differentially expressed genes with the eggNOG annotations

codex <- read.csv("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-11/diff_expr_analysis/OXR-11_eggNOG_codex.txt", sep = " ", header = F, row.names = 1)

new2 <- cbind(data[sort(row.names(data)),],tpm3[sort(row.names(tpm3)),],dta3[sort(row.names(dta3)),],codex[sort(row.names(codex)),])

colnames(new2)[13] <- "eggNOG_alt_name"

new2$eggNOG_name <- row.names(new2)

new2 <- new2[,c("eggNOG_name","eggNOG_alt_name","baseMean","log2FoldChange","lfcSE","stat","pvalue","padj","TPM-C1B3F1","TPM-C1B3F2","TPM-C1B3F3","COUNT-C1B3F1","COUNT-C1B3F2","COUNT-C1B3F3")]

# Load the eggNOG annotation file that was parsed from the "DESeq_diff_expr_from_eggNOG.sh" script
anno <- read.csv("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-11/diff_expr_analysis/OXR-11_diff_expr_annotations.txt", sep = "\t", header = T)

all <- left_join(new2, anno, by = join_by(eggNOG_alt_name == X.query))

output.file <- file("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-11/diff_expr_analysis/OXR-11_annos_w-ALL.txt", "wb")
write.table(all, file = output.file, row.names = F, col.names = T, sep = "\t")
close(output.file)
```

SKIPPING OXR-76

```{r}
## OXR-85
###-------- load packages you need --------###
library(DESeq2)
# library(DEVis)
library(dplyr)

###-------- load data process data --------###
dta <- read.csv("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-85/OXR-85_expt.ct.txt", sep = "\t",
                header=TRUE, row.names = "gene_id")

# remove rows with all zeros
dta <- dta[rowSums(dta[])>0,]
sort(colSums(dta))
sum(colSums(dta))

###-------- define data structure --------###
samples <- data.frame(row.names=c("C1B9F1","C1B9F2","C1B9F3",
                                  "C1B12F1","C1B12F2","C1B12F3",
                                  "C2B1F1","C2B1F2","C2B1F3"), 
                      condition=as.factor(c(rep("C1B9",3),
                                            # rep("C1B12",3),rep("C2B1",3))))
                                            rep("Out",6))))
### Get all samples in the same order
all(rownames(samples) %in% colnames(dta))
all(rownames(samples) == colnames(dta))
dta <- dta[, rownames(samples)]
all(rownames(samples) == colnames(dta))
### Make data integers
dta2 <- as.data.frame(lapply(dta, as.integer))
rownames(dta2) <- rownames(dta)
dta2 <- dta2[rowSums(dta2) > 0,] # Remove more rows with 0's
colSums(dta2)
###-------- create deseq object and process --------###
dta2["fakey",] <- rep(1, ncol(dta2))

cds <- DESeqDataSetFromMatrix(countData = dta2, colData=samples, design=~condition)
cds1 <- DESeq(cds)

ref <- results(cds1)
ref[order(ref$padj),]

c1b9vOut <- results(cds1, contrast=c("condition","C1B9","Out"))
data <- as.data.frame(c1b9vOut[order(c1b9vOut$padj),])
# View(as.data.frame(c1b3vOut[order(c1b3vOut$padj),]))

data <- subset(data, padj <= 0.1)

output.file <- file("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-85/diff_expr_analysis/OXR-85_DESeq_padj.txt", "wb")
write.table(data, file = output.file, row.names = T, col.names = T)
close(output.file)

ids <- row.names(data) %>% as.factor()
ids <- as.factor(ids)
# write.table(ids, "C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-85/OXR-85_padj_trans.txt", row.names = F, col.names = F)
output.file <- file("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-85/diff_expr_analysis/OXR-85_padj_trans.txt", "wb")
write.table(ids, file = output.file, row.names = F, col.names = F)
close(output.file)


#####
# Add TPM data to the "data" data frame
tpm <- read.csv("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-85/OXR-85_TPM.txt", sep = "\t",
                header=TRUE, row.names = "gene_id")
tpm2 <- tpm[,c("C1B9F1", "C1B9F2","C1B9F3")] 
tpm3 <- tpm2[row.names(tpm2) %in% row.names(data),]
names(tpm3) <- c("TPM-C1B9F1","TPM-C1B9F2","TPM-C1B9F3")

# Add the count Data 
dta2 <- dta[,c("C1B9F1","C1B9F2", "C1B9F3")] 
dta3 <- dta2[row.names(dta2) %in% row.names(data),]
names(dta3) <- c("COUNT-C1B9F1","COUNT-C1B9F2","COUNT-C1B9F3")

new <- cbind(data[sort(row.names(data)),],tpm3[sort(row.names(tpm3)),],dta3[sort(row.names(dta3)),])

# Can't do this until the "OXR-X_padj_trans.txt" is run through the bash script "DESeq_diff_expr_from_eggNOG.sh" which collects the differentially expressed genes with the eggNOG annotations

codex <- read.csv("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-85/diff_expr_analysis/OXR-85_eggNOG_codex.txt", sep = " ", header = F, row.names = 1)

new2 <- cbind(data[sort(row.names(data)),],tpm3[sort(row.names(tpm3)),],dta3[sort(row.names(dta3)),],codex[sort(row.names(codex)),])

colnames(new2)[13] <- "eggNOG_alt_name"

new2$eggNOG_name <- row.names(new2)

new2 <- new2[,c("eggNOG_name","eggNOG_alt_name","baseMean","log2FoldChange","lfcSE","stat","pvalue","padj","TPM-C1B9F1","TPM-C1B9F2","TPM-C1B9F3","COUNT-C1B9F1","COUNT-C1B9F2","COUNT-C1B9F3")]

# Load the eggNOG annotation file that was parsed from the "DESeq_diff_expr_from_eggNOG.sh" script
anno <- read.csv("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-85/diff_expr_analysis/OXR-85_diff_expr_annotations.txt", sep = "\t", header = T)

all <- left_join(new2, anno, by = join_by(eggNOG_alt_name == X.query))

output.file <- file("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-85/diff_expr_analysis/OXR-85_annos_w-ALL.txt", "wb")
write.table(all, file = output.file, row.names = T, col.names = T, sep = "\t")
close(output.file)
```

```{r}
## OXR-96
###-------- load packages you need --------###
library(DESeq2)
# library(DEVis)
library(dplyr)

###-------- load data process data --------###
dta <- read.csv("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-96/OXR-96_expt.ct.txt", sep = "\t",
                header=TRUE, row.names = "gene_id")

# remove rows with all zeros
dta <- dta[rowSums(dta[])>0,]
sort(colSums(dta))
sum(colSums(dta))

###-------- define data structure --------###
samples <- data.frame(row.names=c("C1B1F2", "C1B1F1","C1B1F3", # 
                                  "C1B12F1","C1B12F2", "C1B12F3",
                                  "C2B1F1","C2B1F2", "C2B1F3"), 
                      condition=as.factor(c(rep("C1B3",3),
                                            # rep("C1B12",3),rep("C2B1",3))))
                                            rep("Out",6))))
### Get all samples in the same order
all(rownames(samples) %in% colnames(dta))
all(rownames(samples) == colnames(dta))
dta <- dta[, rownames(samples)]
all(rownames(samples) == colnames(dta))
### Make data integers
dta2 <- as.data.frame(lapply(dta, as.integer))
rownames(dta2) <- rownames(dta)
dta2 <- dta2[rowSums(dta2) > 0,] # Remove more rows with 0's
colSums(dta2)
###-------- create deseq object and process --------###
dta2["fakey",] <- rep(1, ncol(dta2))

cds <- DESeqDataSetFromMatrix(countData = dta2, colData=samples, design=~condition)
cds1 <- DESeq(cds)

ref <- results(cds1)
ref[order(ref$padj),]

c1b1vOut <- results(cds1, contrast=c("condition","C1B3","Out"))
data <- as.data.frame(c1b1vOut[order(c1b1vOut$padj),])
# View(as.data.frame(c1b3vOut[order(c1b3vOut$padj),]))

data <- subset(data, padj <= 0.1)

output.file <- file("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-96/diff_expr_analysis/OXR-96_DESeq_padj.txt", "wb")
write.table(data, file = output.file, row.names = T, col.names = T)
close(output.file)

ids <- row.names(data) %>% as.factor()
ids <- as.factor(ids)
# write.table(ids, "C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-96/OXR-96_padj_trans.txt", row.names = F, col.names = F)
output.file <- file("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-96/diff_expr_analysis//OXR-96_padj_trans.txt", "wb")
write.table(ids, file = output.file, row.names = F, col.names = F)
close(output.file)

#####
# Add TPM data to the "data" data frame
tpm <- read.csv("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-96/OXR-96_TPM.txt", sep = "\t",
                header=TRUE, row.names = "gene_id")
tpm2 <- tpm[,c("C1B1F2", "C1B1F1","C1B1F3")] 
tpm3 <- tpm2[row.names(tpm2) %in% row.names(data),]
names(tpm3) <- c("TPM-C1B1F1","TPM-C1B1F2","TPM-C1B1F3")

# Add the count Data 
dta2 <- dta[,c("C1B1F1","C1B1F2", "C1B1F3")] 
dta3 <- dta2[row.names(dta2) %in% row.names(data),]
names(dta3) <- c("COUNT-C1B1F1","COUNT-C1B1F2","COUNT-C1B1F3")

new <- cbind(data[sort(row.names(data)),],tpm3[sort(row.names(tpm3)),],dta3[sort(row.names(dta3)),])

# Can't do this until the "OXR-X_padj_trans.txt" is run through the bash script "DESeq_diff_expr_from_eggNOG.sh" which collects the differentially expressed genes with the eggNOG annotations

codex <- read.csv("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-96/diff_expr_analysis/OXR-96_eggNOG_codex.txt", sep = " ", header = F, row.names = 1)

new2 <- cbind(data[sort(row.names(data)),],tpm3[sort(row.names(tpm3)),],dta3[sort(row.names(dta3)),],codex[sort(row.names(codex)),])

colnames(new2)[13] <- "eggNOG_alt_name"

new2$eggNOG_name <- row.names(new2)

new2 <- new2[,c("eggNOG_name","eggNOG_alt_name","baseMean","log2FoldChange","lfcSE","stat","pvalue","padj","TPM-C1B9F1","TPM-C1B1F2","TPM-C1B1F3","COUNT-C1B1F1","COUNT-C1B1F2","COUNT-C1B1F3")]

# Load the eggNOG annotation file that was parsed from the "DESeq_diff_expr_from_eggNOG.sh" script
anno <- read.csv("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-96/diff_expr_analysis/OXR-96_diff_expr_annotations.txt", sep = "\t", header = T)

all <- left_join(new2, anno, by = join_by(eggNOG_alt_name == X.query))

output.file <- file("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-96/diff_expr_analysis/OXR-96_annos_w-ALL.txt", "wb")
write.table(all, file = output.file, row.names = T, col.names = T, sep = "\t")
close(output.file)
```

```{r}
## OXR-134
###-------- load packages you need --------###
library(DESeq2)
# library(DEVis)
library(dplyr)

###-------- load data process data --------###
dta <- read.csv("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-134/OXR-134_expt.ct.txt", sep = "\t",
                header=TRUE, row.names = "gene_id")

# remove rows with all zeros
dta <- dta[rowSums(dta[])>0,]
sort(colSums(dta))
sum(colSums(dta))

###-------- define data structure --------###
samples <- data.frame(row.names=c( "C1B1F1","C1B1F3", "C1B1F2",
                                  "C1B12F1","C1B12F2","C1B12F3",
                                  "C2B1F1","C2B1F2","C2B1F3"), 
                      condition=as.factor(c(rep("C1B1",3),
                                            # rep("C1B12",3),rep("C2B1",3))))
                                            rep("Out",6))))
### Get all samples in the same order
all(rownames(samples) %in% colnames(dta))
all(rownames(samples) == colnames(dta))
dta <- dta[, rownames(samples)]
all(rownames(samples) == colnames(dta))
### Make data integers
dta2 <- as.data.frame(lapply(dta, as.integer))
rownames(dta2) <- rownames(dta)
dta2 <- dta2[rowSums(dta2) > 0,] # Remove more rows with 0's
colSums(dta2)
###-------- create deseq object and process --------###
dta2["fakey",] <- rep(1, ncol(dta2))

cds <- DESeqDataSetFromMatrix(countData = dta2, colData=samples, design=~condition)
cds1 <- DESeq(cds)

ref <- results(cds1)
ref[order(ref$padj),]

c1b1vOut <- results(cds1, contrast=c("condition","C1B1","Out"))
data <- as.data.frame(c1b1vOut[order(c1b1vOut$padj),])
# View(as.data.frame(c1b3vOut[order(c1b3vOut$padj),]))

data <- subset(data, padj <= 0.1)

output.file <- file("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-134/diff_expr_analysis/OXR-134_DESeq_padj.txt", "wb")
write.table(data, file = output.file, row.names = T, col.names = T)
close(output.file)

ids <- row.names(data) %>% as.factor()
ids <- as.factor(ids)
# write.table(ids, "C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-134/OXR-134_padj_trans.txt", row.names = F, col.names = F)
output.file <- file("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-134/diff_expr_analysis//OXR-134_padj_trans.txt", "wb")
write.table(ids, file = output.file, row.names = F, col.names = F)
close(output.file)

#####
# Add TPM data to the "data" data frame
tpm <- read.csv("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-134/OXR-134_TPM.txt", sep = "\t",
                header=TRUE, row.names = "gene_id")
tpm2 <- tpm[,c("C1B1F2", "C1B1F1","C1B1F3")] 
tpm3 <- tpm2[row.names(tpm2) %in% row.names(data),]
names(tpm3) <- c("TPM-C1B1F1","TPM-C1B1F2","TPM-C1B1F3")

# Add the count Data 
dta2 <- dta[,c("C1B1F1","C1B1F2", "C1B1F3")] 
dta3 <- dta2[row.names(dta2) %in% row.names(data),]
names(dta3) <- c("COUNT-C1B1F1","COUNT-C1B1F2","COUNT-C1B1F3")

new <- cbind(data[sort(row.names(data)),],tpm3[sort(row.names(tpm3)),],dta3[sort(row.names(dta3)),])

# Can't do this until the "OXR-X_padj_trans.txt" is run through the bash script "DESeq_diff_expr_from_eggNOG.sh" which collects the differentially expressed genes with the eggNOG annotations

codex <- read.csv("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-134/diff_expr_analysis/OXR-134_eggNOG_codex.txt", sep = " ", header = F, row.names = 1)

new2 <- cbind(data[sort(row.names(data)),],tpm3[sort(row.names(tpm3)),],dta3[sort(row.names(dta3)),],codex[sort(row.names(codex)),])

colnames(new2)[13] <- "eggNOG_alt_name"

new2$eggNOG_name <- row.names(new2)

new2 <- new2[,c("eggNOG_name","eggNOG_alt_name","baseMean","log2FoldChange","lfcSE","stat","pvalue","padj","TPM-C1B1F1","TPM-C1B1F2","TPM-C1B1F3","COUNT-C1B1F1","COUNT-C1B1F2","COUNT-C1B1F3")]

# Load the eggNOG annotation file that was parsed from the "DESeq_diff_expr_from_eggNOG.sh" script
anno <- read.csv("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-134/diff_expr_analysis/OXR-134_diff_expr_annotations.txt", sep = "\t", header = T)

all <- left_join(new2, anno, by = join_by(eggNOG_alt_name == X.query))

output.file <- file("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-134/diff_expr_analysis/OXR-134_annos_w-ALL.txt", "wb")
write.table(all, file = output.file, row.names = F, col.names = T, sep = "\t")
close(output.file)

```

```{r}
## OXR-137
###-------- load packages you need --------###
library(DESeq2)
# library(DEVis)
library(dplyr)

###-------- load data process data --------###
dta <- read.csv("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-137/OXR-137_expt.ct.txt", sep = "\t",
                header=TRUE, row.names = "gene_id")

# remove rows with all zeros
dta <- dta[rowSums(dta[])>0,]
sort(colSums(dta))
sum(colSums(dta))

###-------- define data structure --------###
samples <- data.frame(row.names=c( "C1B8F1","C1B8F3", "C1B8F2",
                                  "C1B12F1","C1B12F2","C2B1F1",
                                  "C2B1F2", "C1B12F3","C2B1F3"), 
                      condition=as.factor(c(rep("C1B1",3),
                                            rep("Out",6))))
### Get all samples in the same order
all(rownames(samples) %in% colnames(dta))
all(rownames(samples) == colnames(dta))
dta <- dta[, rownames(samples)]
all(rownames(samples) == colnames(dta))
### Make data integers
dta2 <- as.data.frame(lapply(dta, as.integer))
rownames(dta2) <- rownames(dta)
dta2 <- dta2[rowSums(dta2) > 0,] # Remove more rows with 0's
colSums(dta2)
###-------- create deseq object and process --------###
dta2["fakey",] <- rep(1, ncol(dta2))

cds <- DESeqDataSetFromMatrix(countData = dta2, colData=samples, design=~condition)
cds1 <- DESeq(cds)

ref <- results(cds1)
ref[order(ref$padj),]

c1b1vOut <- results(cds1, contrast=c("condition","C1B1","Out"))
data <- as.data.frame(c1b1vOut[order(c1b1vOut$padj),])
# View(as.data.frame(c1b3vOut[order(c1b3vOut$padj),]))

data <- subset(data, padj <= 0.1)

output.file <- file("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-137/diff_expr_analysis/OXR-137_DESeq_padj.txt", "wb")
write.table(data, file = output.file, row.names = T, col.names = T)
close(output.file)

ids <- row.names(data) %>% as.factor()
ids <- as.factor(ids)
# write.table(ids, "C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-137/OXR-137_padj_trans.txt", row.names = F, col.names = F)
output.file <- file("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-137/diff_expr_analysis//OXR-137_padj_trans.txt", "wb")
write.table(ids, file = output.file, row.names = F, col.names = F)
close(output.file)

#####
# Add TPM data to the "data" data frame
tpm <- read.csv("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-137/OXR-137_TPM.txt", sep = "\t",
                header=TRUE, row.names = "gene_id")
tpm2 <- tpm[,c("C1B8F2", "C1B8F1","C1B8F3")] 
tpm3 <- tpm2[row.names(tpm2) %in% row.names(data),]
names(tpm3) <- c("TPM-C1B8F1","TPM-C1B8F2","TPM-C1B8F3")

# Add the count Data 
dta2 <- dta[,c("C1B8F1","C1B8F2", "C1B8F3")] 
dta3 <- dta2[row.names(dta2) %in% row.names(data),]
names(dta3) <- c("COUNT-C1B8F1","COUNT-C1B8F2","COUNT-C1B8F3")

new <- cbind(data[sort(row.names(data)),],tpm3[sort(row.names(tpm3)),],dta3[sort(row.names(dta3)),])

# Can't do this until the "OXR-X_padj_trans.txt" is run through the bash script "DESeq_diff_expr_from_eggNOG.sh" which collects the differentially expressed genes with the eggNOG annotations

codex <- read.csv("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-137/diff_expr_analysis/OXR-137_eggNOG_codex.txt", sep = " ", header = F, row.names = 1)

new2 <- cbind(data[sort(row.names(data)),],tpm3[sort(row.names(tpm3)),],dta3[sort(row.names(dta3)),],codex[sort(row.names(codex)),])

colnames(new2)[13] <- "eggNOG_alt_name"

new2$eggNOG_name <- row.names(new2)

new2 <- new2[,c("eggNOG_name","eggNOG_alt_name","baseMean","log2FoldChange","lfcSE","stat","pvalue","padj","TPM-C1B8F1","TPM-C1B8F2","TPM-C1B8F3","COUNT-C1B8F1","COUNT-C1B8F2","COUNT-C1B8F3")]

# Load the eggNOG annotation file that was parsed from the "DESeq_diff_expr_from_eggNOG.sh" script
anno <- read.csv("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-137/diff_expr_analysis/OXR-137_diff_expr_annotations.txt", sep = "\t", header = T)

all <- left_join(new2, anno, by = join_by(eggNOG_alt_name == X.query))

output.file <- file("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-137/diff_expr_analysis/OXR-137_annos_w-ALL.txt", "wb")
write.table(all, file = output.file, row.names = F, col.names = T, sep = "\t")
close(output.file)

```

```{r}
## OXR-159
###-------- load packages you need --------###
library(DESeq2)
# library(DEVis)
library(dplyr)

###-------- load data process data --------###
dta <- read.csv("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-159/OXR-159_expt.ct.txt", sep = "\t",
                header=TRUE, row.names = "gene_id")

# remove rows with all zeros
dta <- dta[rowSums(dta[])>0,]
sort(colSums(dta))
sum(colSums(dta))

###-------- define data structure --------###
samples <- data.frame(row.names=c( "C1B3F1","C1B3F3", "C1B3F2", # Omit because this was 0
                                  "C1B12F1","C1B12F2","C1B12F3",
                                  "C2B1F2","C2B1F1","C2B1F3"), 
                      condition=as.factor(c(rep("C1B3",3),
                                            # rep("C1B12",3),rep("C2B1",3))))
                                            rep("Out",6))))
### Get all samples in the same order
all(rownames(samples) %in% colnames(dta))
all(rownames(samples) == colnames(dta))
dta <- dta[, rownames(samples)]
all(rownames(samples) == colnames(dta))
### Make data integers
dta2 <- as.data.frame(lapply(dta, as.integer))
rownames(dta2) <- rownames(dta)
dta2 <- dta2[rowSums(dta2) > 0,] # Remove more rows with 0's
colSums(dta2)
###-------- create deseq object and process --------###
dta2["fakey",] <- rep(1, ncol(dta2))

cds <- DESeqDataSetFromMatrix(countData = dta2, colData=samples, design=~condition)
cds1 <- DESeq(cds)

ref <- results(cds1)
ref[order(ref$padj),]

c1b1vOut <- results(cds1, contrast=c("condition","C1B3","Out"))
data <- as.data.frame(c1b1vOut[order(c1b1vOut$padj),])
# View(as.data.frame(c1b3vOut[order(c1b3vOut$padj),]))

data <- subset(data, padj <= 0.1)

output.file <- file("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-159/diff_expr_analysis/OXR-159_DESeq_padj.txt", "wb")
write.table(data, file = output.file, row.names = T, col.names = T)
close(output.file)

ids <- row.names(data) %>% as.factor()
ids <- as.factor(ids)
# write.table(ids, "C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-159/OXR-159_padj_trans.txt", row.names = F, col.names = F)
output.file <- file("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-159/diff_expr_analysis//OXR-159_padj_trans.txt", "wb")
write.table(ids, file = output.file, row.names = F, col.names = F)
close(output.file)

#####
# Add TPM data to the "data" data frame
tpm <- read.csv("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-159/OXR-159_TPM.txt", sep = "\t",
                header=TRUE, row.names = "gene_id")
tpm2 <- tpm[,c("C1B3F1", "C1B3F2","C1B3F3")] 
tpm3 <- tpm2[row.names(tpm2) %in% row.names(data),]
names(tpm3) <- c("TPM-C1B3F1","TPM-C1B3F2","TPM-C1B3F3")

# Add the count Data 
dta2 <- dta[,c("C1B3F1","C1B3F2", "C1B3F3")] 
dta3 <- dta2[row.names(dta2) %in% row.names(data),]
names(dta3) <- c("COUNT-C1B3F1","COUNT-C1B3F2","COUNT-C1B3F3")

new <- cbind(data[sort(row.names(data)),],tpm3[sort(row.names(tpm3)),],dta3[sort(row.names(dta3)),])

# Can't do this until the "OXR-X_padj_trans.txt" is run through the bash script "DESeq_diff_expr_from_eggNOG.sh" which collects the differentially expressed genes with the eggNOG annotations

codex <- read.csv("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-159/diff_expr_analysis/OXR-159_eggNOG_codex.txt", sep = " ", header = F, row.names = 1)

new2 <- cbind(data[sort(row.names(data)),],tpm3[sort(row.names(tpm3)),],dta3[sort(row.names(dta3)),],codex[sort(row.names(codex)),])

colnames(new2)[13] <- "eggNOG_alt_name"

new2$eggNOG_name <- row.names(new2)

new2 <- new2[,c("eggNOG_name","eggNOG_alt_name","baseMean","log2FoldChange","lfcSE","stat","pvalue","padj","TPM-C1B3F1","TPM-C1B3F2","TPM-C1B3F3","COUNT-C1B3F1","COUNT-C1B3F2","COUNT-C1B3F3")]

# Load the eggNOG annotation file that was parsed from the "DESeq_diff_expr_from_eggNOG.sh" script
anno <- read.csv("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-159/diff_expr_analysis/OXR-159_diff_expr_annotations.txt", sep = "\t", header = T)

all <- left_join(new2, anno, by = join_by(eggNOG_alt_name == X.query))

output.file <- file("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-159/diff_expr_analysis/OXR-159_annos_w-ALL.txt", "wb")
write.table(all, file = output.file, row.names = F, col.names = T, sep = "\t")
close(output.file)

```

```{r}
## OXR-189
###-------- load packages you need --------###
library(DESeq2)
# library(DEVis)
library(dplyr)

###-------- load data process data --------###
dta <- read.csv("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-189/OXR-189_expt.ct.txt", sep = "\t",
                header=TRUE, row.names = "gene_id")

# remove rows with all zeros
dta <- dta[rowSums(dta[])>0,]
sort(colSums(dta))
sum(colSums(dta))

###-------- define data structure --------###
samples <- data.frame(row.names=c( "C1B8F1","C1B8F3", "C1B8F2",
                                  "C1B12F1","C1B12F2","C1B12F3",
                                  "C2B1F1","C2B1F2","C2B1F3"), 
                      condition=as.factor(c(rep("C1B8",3),
                                            # rep("C1B12",3),rep("C2B1",3))))
                                            rep("Out",6))))
### Get all samples in the same order
all(rownames(samples) %in% colnames(dta))
all(rownames(samples) == colnames(dta))
dta <- dta[, rownames(samples)]
all(rownames(samples) == colnames(dta))
### Make data integers
dta2 <- as.data.frame(lapply(dta, as.integer))
rownames(dta2) <- rownames(dta)
dta2 <- dta2[rowSums(dta2) > 0,] # Remove more rows with 0's
colSums(dta2)
###-------- create deseq object and process --------###
dta2["fakey",] <- rep(1, ncol(dta2))

cds <- DESeqDataSetFromMatrix(countData = dta2, colData=samples, design=~condition)
cds1 <- DESeq(cds)

ref <- results(cds1)
ref[order(ref$padj),]

c1b1vOut <- results(cds1, contrast=c("condition","C1B8","Out"))
data <- as.data.frame(c1b1vOut[order(c1b1vOut$padj),])
# View(as.data.frame(c1b3vOut[order(c1b3vOut$padj),]))

data <- subset(data, padj <= 0.1)

output.file <- file("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-189/diff_expr_analysis/OXR-189_DESeq_padj.txt", "wb")
write.table(data, file = output.file, row.names = T, col.names = T)
close(output.file)

ids <- row.names(data) %>% as.factor()
ids <- as.factor(ids)
# write.table(ids, "C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-189/OXR-189_padj_trans.txt", row.names = F, col.names = F)
output.file <- file("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-189/diff_expr_analysis//OXR-189_padj_trans.txt", "wb")
write.table(ids, file = output.file, row.names = F, col.names = F)
close(output.file)

#####
# Add TPM data to the "data" data frame
tpm <- read.csv("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-189/OXR-189_TPM.txt", sep = "\t",
                header=TRUE, row.names = "gene_id")
tpm2 <- tpm[,c("C1B8F1", "C1B8F2","C1B8F3")] 
tpm3 <- tpm2[row.names(tpm2) %in% row.names(data),]
names(tpm3) <- c("TPM-C1B8F1","TPM-C1B8F2","TPM-C1B8F3")

# Add the count Data 
dta2 <- dta[,c("C1B8F1","C1B8F2", "C1B8F3")] 
dta3 <- dta2[row.names(dta2) %in% row.names(data),]
names(dta3) <- c("COUNT-C1B8F1","COUNT-C1B8F2","COUNT-C1B8F3")

new <- cbind(data[sort(row.names(data)),],tpm3[sort(row.names(tpm3)),],dta3[sort(row.names(dta3)),])

# Can't do this until the "OXR-X_padj_trans.txt" is run through the bash script "DESeq_diff_expr_from_eggNOG.sh" which collects the differentially expressed genes with the eggNOG annotations

codex <- read.csv("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-189/diff_expr_analysis/OXR-189_eggNOG_codex.txt", sep = " ", header = F, row.names = 1)

new2 <- cbind(data[sort(row.names(data)),],tpm3[sort(row.names(tpm3)),],dta3[sort(row.names(dta3)),],codex[sort(row.names(codex)),])

colnames(new2)[13] <- "eggNOG_alt_name"

new2$eggNOG_name <- row.names(new2)

new2 <- new2[,c("eggNOG_name","eggNOG_alt_name","baseMean","log2FoldChange","lfcSE","stat","pvalue","padj","TPM-C1B8F1","TPM-C1B8F2","TPM-C1B8F3","COUNT-C1B8F1","COUNT-C1B8F2","COUNT-C1B8F3")]

# Load the eggNOG annotation file that was parsed from the "DESeq_diff_expr_from_eggNOG.sh" script
anno <- read.csv("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-189/diff_expr_analysis/OXR-189_diff_expr_annotations.txt", sep = "\t", header = T)

all <- left_join(new2, anno, by = join_by(eggNOG_alt_name == X.query))

output.file <- file("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-189/diff_expr_analysis/OXR-189_annos_w-ALL.txt", "wb")
write.table(all, file = output.file, row.names = F, col.names = T, sep = "\t")
close(output.file)

```

```{r}
## OXR-199
###-------- load packages you need --------###
library(DESeq2)
# library(DEVis)
library(dplyr)

###-------- load data process data --------###
dta <- read.csv("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-199/OXR-199_expt.ct.txt", sep = "\t",
                header=TRUE, row.names = "gene_id")

# remove rows with all zeros
dta <- dta[rowSums(dta[])>0,]
sort(colSums(dta))
sum(colSums(dta))

###-------- define data structure --------###
samples <- data.frame(row.names=c("C1B1F1","C1B1F3", "C1B1F2",
                                  #"C1B2F1","C1B2F3", "C1B2F2",
                                  "C1B12F1","C1B12F2","C1B12F3",
                                  "C2B1F1","C2B1F2", "C2B1F3"), 
                      condition=as.factor(c(rep("C1B1",3),
                                            # rep("C1B12",3),rep("C2B1",3))))
                                            rep("Out",6))))
### Get all samples in the same order
all(rownames(samples) %in% colnames(dta))
all(rownames(samples) == colnames(dta))
dta <- dta[, rownames(samples)]
all(rownames(samples) == colnames(dta))
### Make data integers
dta2 <- as.data.frame(lapply(dta, as.integer))
rownames(dta2) <- rownames(dta)
dta2 <- dta2[rowSums(dta2) > 0,] # Remove more rows with 0's
colSums(dta2)
###-------- create deseq object and process --------###
dta2["fakey",] <- rep(1, ncol(dta2))

cds <- DESeqDataSetFromMatrix(countData = dta2, colData=samples, design=~condition)
cds1 <- DESeq(cds)

ref <- results(cds1)
ref[order(ref$padj),]

c1b1vOut <- results(cds1, contrast=c("condition","C1B1","Out"))
data <- as.data.frame(c1b1vOut[order(c1b1vOut$padj),])
# View(as.data.frame(c1b3vOut[order(c1b3vOut$padj),]))

data <- subset(data, padj <= 0.1)

output.file <- file("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-199/diff_expr_analysis/OXR-199_DESeq_padj.txt", "wb")
write.table(data, file = output.file, row.names = T, col.names = T)
close(output.file)

ids <- row.names(data) %>% as.factor()
ids <- as.factor(ids)
# write.table(ids, "C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-199/OXR-199_padj_trans.txt", row.names = F, col.names = F)
output.file <- file("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-199/diff_expr_analysis//OXR-199_padj_trans.txt", "wb")
write.table(ids, file = output.file, row.names = F, col.names = F)
close(output.file)

#####
# Add TPM data to the "data" data frame
tpm <- read.csv("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-199/OXR-199_TPM.txt", sep = "\t",
                header=TRUE, row.names = "gene_id")
tpm2 <- tpm[,c("C1B1F1", "C1B1F2","C1B1F3")] 
tpm3 <- tpm2[row.names(tpm2) %in% row.names(data),]
names(tpm3) <- c("TPM-C1B1F1","TPM-C1B1F2","TPM-C1B1F3")

# Add the count Data 
dta2 <- dta[,c("C1B1F1","C1B1F2", "C1B1F3")] 
dta3 <- dta2[row.names(dta2) %in% row.names(data),]
names(dta3) <- c("COUNT-C1B1F1","COUNT-C1B1F2","COUNT-C1B1F3")

new <- cbind(data[sort(row.names(data)),],tpm3[sort(row.names(tpm3)),],dta3[sort(row.names(dta3)),])

# Can't do this until the "OXR-X_padj_trans.txt" is run through the bash script "DESeq_diff_expr_from_eggNOG.sh" which collects the differentially expressed genes with the eggNOG annotations

codex <- read.csv("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-199/diff_expr_analysis/OXR-199_eggNOG_codex.txt", sep = " ", header = F, row.names = 1)

new2 <- cbind(data[sort(row.names(data)),],tpm3[sort(row.names(tpm3)),],dta3[sort(row.names(dta3)),],codex[sort(row.names(codex)),])

colnames(new2)[13] <- "eggNOG_alt_name"

new2$eggNOG_name <- row.names(new2)

new2 <- new2[,c("eggNOG_name","eggNOG_alt_name","baseMean","log2FoldChange","lfcSE","stat","pvalue","padj","TPM-C1B1F1","TPM-C1B1F2","TPM-C1B1F3","COUNT-C1B1F1","COUNT-C1B1F2","COUNT-C1B1F3")]

# Load the eggNOG annotation file that was parsed from the "DESeq_diff_expr_from_eggNOG.sh" script
anno <- read.csv("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-199/diff_expr_analysis/OXR-199_diff_expr_annotations.txt", sep = "\t", header = T)

all <- left_join(new2, anno, by = join_by(eggNOG_alt_name == X.query))

output.file <- file("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-199/diff_expr_analysis/OXR-199_annos_w-ALL.txt", "wb")
write.table(all, file = output.file, row.names = F, col.names = T, sep = "\t")
close(output.file)
```

```{r}
## OXR-203
###-------- load packages you need --------###
library(DESeq2)
# library(DEVis)
library(dplyr)

###-------- load data process data --------###
dta <- read.csv("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-203/OXR-203_expt.ct.txt", sep = "\t",
                header=TRUE, row.names = "gene_id")

# remove rows with all zeros
dta <- dta[rowSums(dta[])>0,]
sort(colSums(dta))
sum(colSums(dta))

###-------- define data structure --------###
samples <- data.frame(row.names=c( "C1B1F1","C1B1F3", "C1B1F2",
                                  "C1B12F1","C1B12F2","C1B12F3",
                                  "C2B1F1","C2B1F2","C2B1F3"), 
                      condition=as.factor(c(rep("C1B1",3),
                                            # rep("C1B12",3),rep("C2B1",3))))
                                            rep("Out",6))))
### Get all samples in the same order
all(rownames(samples) %in% colnames(dta))
all(rownames(samples) == colnames(dta))
dta <- dta[, rownames(samples)]
all(rownames(samples) == colnames(dta))
### Make data integers
dta2 <- as.data.frame(lapply(dta, as.integer))
rownames(dta2) <- rownames(dta)
dta2 <- dta2[rowSums(dta2) > 0,] # Remove more rows with 0's
colSums(dta2)
###-------- create deseq object and process --------###
dta2["fakey",] <- rep(1, ncol(dta2))

cds <- DESeqDataSetFromMatrix(countData = dta2, colData=samples, design=~condition)
cds1 <- DESeq(cds)

ref <- results(cds1)
ref[order(ref$padj),]

c1b1vOut <- results(cds1, contrast=c("condition","C1B1","Out"))
data <- as.data.frame(c1b1vOut[order(c1b1vOut$padj),])
# View(as.data.frame(c1b3vOut[order(c1b3vOut$padj),]))

data <- subset(data, padj <= 0.1)

output.file <- file("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-203/diff_expr_analysis/OXR-203_DESeq_padj.txt", "wb")
write.table(data, file = output.file, row.names = T, col.names = T)
close(output.file)

ids <- row.names(data) %>% as.factor()
ids <- as.factor(ids)
# write.table(ids, "C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-203/OXR-203_padj_trans.txt", row.names = F, col.names = F)
output.file <- file("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-203/diff_expr_analysis//OXR-203_padj_trans.txt", "wb")
write.table(ids, file = output.file, row.names = F, col.names = F)
close(output.file)

#####
# Add TPM data to the "data" data frame
tpm <- read.csv("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-203/OXR-203_TPM.txt", sep = "\t",
                header=TRUE, row.names = "gene_id")
tpm2 <- tpm[,c("C1B1F1", "C1B1F2","C1B1F3")] 
tpm3 <- tpm2[row.names(tpm2) %in% row.names(data),]
names(tpm3) <- c("TPM-C1B1F1","TPM-C1B1F2","TPM-C1BF3")

# Add the count Data 
dta2 <- dta[,c("C1B1F1","C1B1F2", "C1B1F3")] 
dta3 <- dta2[row.names(dta2) %in% row.names(data),]
names(dta3) <- c("COUNT-C1B1F1","COUNT-C1B1F2","COUNT-C1B1F3")

new <- cbind(data[sort(row.names(data)),],tpm3[sort(row.names(tpm3)),],dta3[sort(row.names(dta3)),])

# Can't do this until the "OXR-X_padj_trans.txt" is run through the bash script "DESeq_diff_expr_from_eggNOG.sh" which collects the differentially expressed genes with the eggNOG annotations

codex <- read.csv("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-203/diff_expr_analysis/OXR-203_eggNOG_codex.txt", sep = " ", header = F, row.names = 1)

new2 <- cbind(data[sort(row.names(data)),],tpm3[sort(row.names(tpm3)),],dta3[sort(row.names(dta3)),],codex[sort(row.names(codex)),])

colnames(new2)[13] <- "eggNOG_alt_name"

new2$eggNOG_name <- row.names(new2)

new2 <- new2[,c("eggNOG_name","eggNOG_alt_name","baseMean","log2FoldChange","lfcSE","stat","pvalue","padj","TPM-C1B1F1","TPM-C1B1F2","TPM-C1B1F3","COUNT-C1B1F1","COUNT-C1B1F2","COUNT-C1B1F3")]

# Load the eggNOG annotation file that was parsed from the "DESeq_diff_expr_from_eggNOG.sh" script
anno <- read.csv("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-203/diff_expr_analysis/OXR-203_diff_expr_annotations.txt", sep = "\t", header = T)

all <- left_join(new2, anno, by = join_by(eggNOG_alt_name == X.query))

output.file <- file("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-203/diff_expr_analysis/OXR-203_annos_w-ALL.txt", "wb")
write.table(all, file = output.file, row.names = F, col.names = T, sep = "\t")
close(output.file)
```

```{r}
## OXR-209
###-------- load packages you need --------###
library(DESeq2)
# library(DEVis)
library(dplyr)

###-------- load data process data --------###
dta <- read.csv("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-209/OXR-209_expt.ct.txt", sep = "\t",
                header=TRUE, row.names = "gene_id")

# remove rows with all zeros
dta <- dta[rowSums(dta[])>0,]
sort(colSums(dta))
sum(colSums(dta))

###-------- define data structure --------###
samples <- data.frame(row.names=c( "C1B11F1", "C1B11F2", "C1B11F3",
                                  "C1B12F1","C1B12F2","C1B12F3",
                                  "C2B1F1","C2B1F2","C2B1F3"), 
                      condition=as.factor(c(rep("C1B1",3),
                                            # rep("C1B12",3),rep("C2B1",3))))
                                            rep("Out",6))))
### Get all samples in the same order
all(rownames(samples) %in% colnames(dta))
all(rownames(samples) == colnames(dta))
dta <- dta[, rownames(samples)]
all(rownames(samples) == colnames(dta))
### Make data integers
dta2 <- as.data.frame(lapply(dta, as.integer))
rownames(dta2) <- rownames(dta)
dta2 <- dta2[rowSums(dta2) > 0,] # Remove more rows with 0's
colSums(dta2)
###-------- create deseq object and process --------###
dta2["fakey",] <- rep(1, ncol(dta2))

cds <- DESeqDataSetFromMatrix(countData = dta2, colData=samples, design=~condition)
cds1 <- DESeq(cds)

ref <- results(cds1)
ref[order(ref$padj),]

c1b1vOut <- results(cds1, contrast=c("condition","C1B1","Out"))
data <- as.data.frame(c1b1vOut[order(c1b1vOut$padj),])
# View(as.data.frame(c1b3vOut[order(c1b3vOut$padj),]))

data <- subset(data, padj <= 0.1)

output.file <- file("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-209/diff_expr_analysis/OXR-209_DESeq_padj.txt", "wb")
write.table(data, file = output.file, row.names = T, col.names = T)
close(output.file)

ids <- row.names(data) %>% as.factor()
ids <- as.factor(ids)
# write.table(ids, "C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-209/OXR-209_padj_trans.txt", row.names = F, col.names = F)
output.file <- file("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-209/diff_expr_analysis//OXR-209_padj_trans.txt", "wb")
write.table(ids, file = output.file, row.names = F, col.names = F)
close(output.file)

#####
# Add TPM data to the "data" data frame
tpm <- read.csv("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-209/OXR-209_TPM.txt", sep = "\t",
                header=TRUE, row.names = "gene_id")
tpm2 <- tpm[,c("C1B11F1", "C1B11F2","C1B11F3")] 
tpm3 <- tpm2[row.names(tpm2) %in% row.names(data),]
names(tpm3) <- c("TPM-C1B11F1","TPM-C1B11F2","TPM-C1B11F3")

# Add the count Data 
dta2 <- dta[,c("C1B11F1","C1B11F2", "C1B11F3")] 
dta3 <- dta2[row.names(dta2) %in% row.names(data),]
names(dta3) <- c("COUNT-C1B11F1","COUNT-C1B11F2","COUNT-C1B11F3")

new <- cbind(data[sort(row.names(data)),],tpm3[sort(row.names(tpm3)),],dta3[sort(row.names(dta3)),])

# Can't do this until the "OXR-X_padj_trans.txt" is run through the bash script "DESeq_diff_expr_from_eggNOG.sh" which collects the differentially expressed genes with the eggNOG annotations

codex <- read.csv("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-209/diff_expr_analysis/OXR-209_eggNOG_codex.txt", sep = " ", header = F, row.names = 1)

new2 <- cbind(data[sort(row.names(data)),],tpm3[sort(row.names(tpm3)),],dta3[sort(row.names(dta3)),],codex[sort(row.names(codex)),])

colnames(new2)[13] <- "eggNOG_alt_name"

new2$eggNOG_name <- row.names(new2)

new2 <- new2[,c("eggNOG_name","eggNOG_alt_name","baseMean","log2FoldChange","lfcSE","stat","pvalue","padj","TPM-C1B11F1","TPM-C1B11F2","TPM-C1B11F3","COUNT-C1B11F1","COUNT-C1B11F2","COUNT-C1B11F3")]

# Load the eggNOG annotation file that was parsed from the "DESeq_diff_expr_from_eggNOG.sh" script
anno <- read.csv("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-209/diff_expr_analysis/OXR-209_diff_expr_annotations.txt", sep = "\t", header = T)

all <- left_join(new2, anno, by = join_by(eggNOG_alt_name == X.query))

output.file <- file("C:/Users/SamuelMajor/OneDrive - Gloucester Marine Genomics Institute/Desktop/OXR genomes/OXR-209/diff_expr_analysis/OXR-209_annos_w-ALL.txt", "wb")
write.table(all, file = output.file, row.names = F, col.names = T, sep = "\t")
close(output.file)

```